account_invoice_fixed_discount/models/account_invoice.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, _
+import openerp.addons.decimal_precision as dp
+from openerp.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def get_taxes_values(self):
+        self.ensure_one()
+        vals = {}
+        for line in self.invoice_line_ids.filtered('discount_fixed'):
+            vals[line] = {
+                'price_unit': line.price_unit,
+                'discount_fixed': line.discount_fixed,
+            }
+            price_unit = line.price_unit - line.discount_fixed
+            line.update({
+                'price_unit': price_unit,
+                'discount_fixed': 0.0,
+            })
+        tax_grouped = super(AccountInvoice, self).get_taxes_values()
+        for line in vals.keys():
+            line.update(vals[line])
+        return tax_grouped
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    discount_fixed = fields.Float(
+        string="Discount (Fixed)",
+        digits=dp.get_precision('Product Price'),
+        help="Fixed amount discount.")
+
+    @api.onchange('discount')
+    def _onchange_discount(self):
+        if self.discount:
+            self.discount_fixed = 0.0
+
+    @api.onchange('discount_fixed')
+    def _onchange_discount_fixed(self):
+        if self.discount_fixed:
+            self.discount = 0.0
+
+    @api.multi
+    @api.constrains('discount', 'discount_fixed')
+    def _check_only_one_discount(self):
+        for line in self:
+            if line.discount and line.discount_fixed:
+                raise ValidationError(
+                    _("You can only set one type of discount per line."))
+
+    @api.one
+    @api.depends('price_unit', 'discount', 'invoice_line_tax_ids', 'quantity',
+                 'product_id', 'invoice_id.partner_id',
+                 'invoice_id.currency_id', 'invoice_id.company_id',
+                 'discount_fixed')
+    def _compute_price(self):
```

Comment:
@pedrobaeza @lasley 
What do you think about a inherit of a computed method?

Reply:
Well, we do something similar in https://github.com/OCA/account-invoicing/blob/7a2aa5c6ed53cb0e9d853219b04e7224a93bd2ee/account_invoice_triple_discount/models/account_invoice.py#L49, and these cases inheritance will fail, but there's no other way to do it. You can mention the incompatibility in the README.

Reply:
what do you mean? is it impossible to have installed and working fine both this module and yours?

Reply:
Yeah, as both modifies the price_unit as trick for having the rest without taking into account possible previous modifications. You can't have a fixed price discount and also second and third discount. If you see we can find a way to conciliate both....

Reply:
We could try to add a hook module which we both can inherit into our modules. Do you think that would be possible?